master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -64,17 +66,19 @@ public abstract class AbstractMetaManager implements IMetadataHandler {
   public final Map<String, Set<Integer>> registeredAppAndShuffles =
       JavaUtils.newConcurrentHashMap();
   public final Set<String> hostnameSet = ConcurrentHashMap.newKeySet();
-  public final Set<WorkerInfo> workers = ConcurrentHashMap.newKeySet();
+  public final ConcurrentHashMap<String, Long> appHeartbeatTime = JavaUtils.newConcurrentHashMap();
 
-  public final ConcurrentHashMap<WorkerInfo, Long> lostWorkers = JavaUtils.newConcurrentHashMap();
-  public final ConcurrentHashMap<WorkerInfo, WorkerEventInfo> workerEventInfos =
+  private final Map<String, WorkerInfo> workersMap = JavaUtils.newConcurrentHashMap();
```

Comment:
I believe we only need to change workers to workersMap for quicker access to available workers. For other parts, like excludedWorkers and manuallyExcludedWorkers, it might be more efficient to use a Set<WorkInfo>, as there doesn't seem to be a demand for rapid lookups in those cases. Introducing a workerInfoPool might complicate things unnecessarily.

Reply:
Will raise a PR to only reserve the workersMap, thanks